examples/reference/custom_components/PreactComponent.md

```diff
@@ -0,0 +1,408 @@
+        self.value = str(event.__dict__)
+
+button = EventExample()
+pn.Column(
+    button, pn.widgets.TextAreaInput(value=button.param.value, height=200),
+).servable()
+```
+
+You can also define and send your own custom events:
+
+```python
+import panel as pn
+import param
+
+from panel.custom import PreactComponent
+
+pn.extension()
+
+class CustomEventExample(PreactComponent):
+
+    value = param.Parameter()
+
+    _esm = """
+    function send_event(data) {
+        const currentDate = new Date();
+        const custom_event = new CustomEvent("click", { detail: currentDate.getTime() });
+        data.send_event('click', custom_event);
+    }
+
+    function App(props) {
+        return html`
+            <button onClick=${() => send_event(props.data)}>
+                Click me
+            </button>
+        `;
+    }
+
+    export function render({ data }) {
+        return html`<App data=${data} />`
+    }
+    """
+
+def _handle_click(self, event):
+    self.value = str(event.__dict__)
+
+button = CustomEventExample()
+pn.Column(
+    button, pn.widgets.TextAreaInput(value=button.param.value, height=200),
+).servable()
+```
+
+## Dependency Imports
+
+JavaScript dependencies can be directly imported via URLs, such as those from [`esm.sh`](https://esm.sh/).
+
+```python
+import panel as pn
+
+from panel.custom import PreactComponent
+
+pn.extension()
+
+class ConfettiButton(PreactComponent):
+    _esm = """
+    import confetti from "https://esm.sh/canvas-confetti@1.6.0";
+
+    export function render() {
+        return html`
+            <button onClick=${e => confetti()}>
+                Click Me
+            </button>
+        `;
+    }
+    """
+
+ConfettiButton().servable()
+```
+
+Use the `_importmap` attribute for more concise module references.
+
+```python
+import panel as pn
+
+from panel.custom import PreactComponent
+
+pn.extension()
+
+class ConfettiButton(PreactComponent):
+    _importmap = {
+        "imports": {
+            "canvas-confetti": "https://esm.sh/canvas-confetti@1.6.0",
+        }
+    }
+
+    _esm = """
+    import confetti from "canvas-confetti";
+
+    export function render() {
+        return html`
+            <button onClick=${e => confetti()}>
+                Click Me
+            </button>
+        `;
+    }
+    """
+
+ConfettiButton().servable()
+```
+
+See [import map](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type/importmap) for more info.
+
+## External Files
+
+You can load JavaScript and CSS from files by providing the paths to these files.
+
+Create the file **counter_button.py**.
+
+```python
+from pathlib import Path
+
+import param
+import panel as pn
+
+pn.extension()
+
+class CounterButton(pn.PreactComponent):
+
+    value = param.Integer()
+
+    _esm = "counter_button.js"
+    _stylesheets = [Path("counter_button.css")]
+
+CounterButton().servable()
+```
+
+Now create the file **counter_button.js**.
+
+```javascript
+export function render({ data }) {
```

Comment:
Autoreload of the js file no longer works for me as described for the `ReactComponent`.